src/frontend/Komponenter/Behandling/VedtakOgBeregning/Barnetilsyn/InnvilgeVedtak/InnvilgeBarnetilsyn.tsx

```diff
@@ -175,13 +173,13 @@ export const InnvilgeBarnetilsyn: React.FC<{
     useEffect(() => {
         if (
             behandlingErRedigerbar &&
-            finnesKontantstøtteUtbetaling === FinnesKontantstøtteUtbetaling.NEI &&
+            harKontantstøttePerioder === false &&
             kontantstøtteState.value === ERadioValg.IKKE_SATT
         ) {
             kontantstøtteState.setValue(ERadioValg.NEI);
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [finnesKontantstøtteUtbetaling]);
+    }, [harKontantstøttePerioder]);
```

Comment:
Kan dette dyttes rett inn i `initKontantstøttestate` nå - siden vi har dataene tilgjengelig?

Da kan hele useEffecten fjernes isåfall

Reply:
```
const initKontantstøttestate = (vedtak: IInnvilgeVedtakForBarnetilsyn | undefined) =>
    vedtak
        ? vedtak.perioderKontantstøtte && vedtak.perioderKontantstøtte.length > 0
            ? ERadioValg.JA
            : ERadioValg.NEI
        : ERadioValg.IKKE_SATT;
```

kan f.eks bli til 
```
const initKontantstøttestate = (vedtak: IInnvilgeVedtakForBarnetilsyn | undefined) =>
    vedtak
        ? vedtak.perioderKontantstøtte && vedtak.perioderKontantstøtte.length > 0
            ? ERadioValg.JA
            : ERadioValg.NEI
        : ( behandlingErRedigerbar &&
            harKontantstøttePerioder === false) ? ERadioValg.NEI : ERadioValg.IKKE_SATT;
```

(kanskje litt penere, men du skjønner kanskje greia?